features/device-config/persistence/impl/src/main/java/org/opennms/features/deviceconfig/persistence/impl/DeviceConfigDaoImpl.java

```diff
@@ -249,6 +253,25 @@ public List<DeviceConfig> getAllDeviceConfigsWithAnInterfaceId(Integer ipInterfa
                 ipInterfaceId);
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public Map<String, Long> getNumberOfNodesWithConfigBySysOid() {
+        String query = "select n.nodesysoid, count(*) from device_config dcb LEFT JOIN ipinterface ip ON ipinterface_id = ip.id LEFT JOIN node n ON ip.nodeid = n.nodeid group by nodesysoid";
+
+        List<Object[]> pairs = getHibernateTemplate().executeWithNativeSession(session -> {
+             SQLQuery queryObject = session.createSQLQuery(query);
+             return queryObject.list();
+
+        });
+        Map<String, Long> numberOfNodesWithConfigBySysOid = new HashMap<String, Long>();
```

Comment:
This can/should be reworked to build the map in the transaction using `queryObject.iterate()`.

Reply:
Updated to construct the map inside the transaction. Using an iterator gave me an `UnsupportedOperationException` saying that native SQL queries don't support iterators, so I used `list()` instead.